bot/cogs/eval.py

```diff
@@ -12,66 +19,153 @@
 class EvalCog:  # Named this way because a flake8 plugin isn't case-sensitive
     """
     Bot owner only: Evaluate Python code
+    
+    Made by martmists
     """
 
     def __init__(self, bot: AutoShardedBot):
         self.bot = bot
+        self.env = {}
+        self.ln = 0
+        self.stdout = io.StringIO()
+
         self.interpreter = Interpreter(bot)
+    
+    def _format(self, inp, out):
+        self._ = out
 
-    @command()
-    @with_role(OWNER_ROLE)
-    async def eval(self, ctx: Context, *, string: str):
-        """
-        Bot owner only: Evaluate Python code
+        res = ""
 
-        Your code may be surrounded in a code fence, but it's not required.
-        Scope will be preserved - variables set will be present later on.
-        """
+        # Erase temp input we made
+        if inp.startswith("_ = "):
+            inp = inp[4:]
 
-        code = string.strip()
+        lines = [l for l in inp.split("\n") if l.strip()]
+        if len(lines) != 1:
+            lines += [""]
+
+        # Create the inpit dialog
+        for i, line in enumerate(lines):
+            if i == 0:
+                s = f"In [{self.ln}]: "
 
-        if code.startswith("```") and code.endswith("```"):
-            if code.startswith("```python"):
-                code = code[9:-3]
-            elif code.startswith("```py"):
-                code = code[5:-3]
             else:
-                code = code[3:-3]
-        elif code.startswith("`") and code.endswith("`"):
-            code = code[1:-1]
+                # Indent the 3 dots correctly
+                s = (f"{{:<{len(str(self.ln))+2}}}...: ").format("")
```

Comment:
Can you comment this better? It's very unclear what this would do for a beginner